cpp/oneapi/dal/backend/primitives/selection/kselect_by_rows_simd.hpp

```diff
@@ -85,40 +83,36 @@ class kselect_by_rows_simd : public kselect_by_rows_base<Float> {
         const std::int64_t stride = data.get_shape()[1];
 
         const std::int64_t row_adjusted_sg_num =
-            col_count / sg_max_size + (std::int64_t)((bool)(col_count % sg_max_size));
+            col_count / sg_max_size + std::int64_t(col_count % sg_max_size > 0);
         const std::int64_t expected_sg_num =
             std::min(preffered_wg_size / sg_max_size, row_adjusted_sg_num);
         ONEDAL_ASSERT(expected_sg_num > 0);
```

Comment:
This can be extracted in the separate method on calculating expected subgroup count